sru/sru_cpu_impl.cpp

```diff
@@ -0,0 +1,240 @@
+
+#include <torch/torch.h>
+#include <vector>
+
+std::vector<at::Tensor> cpu_forward(
+        const at::Tensor & U,
+        const at::Tensor & x,
+        const at::Tensor & weight_c,
+        const at::Tensor & bias,
+        const at::Tensor & c_init,
+        const at::Tensor & mask_pad,
+        int64_t length, 
+        int64_t batch_size, 
+        int64_t d, 
+        int64_t k,
+        int64_t activation_type,
+        bool has_skip_term, 
+        double scale_x);
+
+std::vector<at::Tensor> cpu_bi_forward(
+        const at::Tensor & U,
+        const at::Tensor & x,
+        const at::Tensor & weight_c,
+        const at::Tensor & bias,
+        const at::Tensor & c_init,
+        const at::Tensor & mask_pad,
+        int64_t length, 
+        int64_t batch_size, 
+        int64_t d, 
+        int64_t k,
+        int64_t activation_type,
+        bool has_skip_term, 
+        double scale_x);
+
+float sigmoidf(float x);
+float reluf(float x);
+float seluf(float x);
+float apply_activation(int64_t type, float x);
+
+
+/* Implementation starts here */
+
+#define CHECK_CONTIGUOUS(x) AT_ASSERTM(x.is_contiguous(), #x " must be contiguous")
+#define CHECK_FLOAT(x) AT_ASSERTM(x.type().scalarType() == at::ScalarType::Float, #x " must be float")
+
+
+std::vector<at::Tensor> cpu_forward(
+        const at::Tensor & U,
+        const at::Tensor & x,
+        const at::Tensor & weight_c,
+        const at::Tensor & bias,
+        const at::Tensor & c_init,
+        const at::Tensor & mask_pad,
+        int64_t length, 
+        int64_t batch_size, 
+        int64_t d, 
```

Comment:
what is d? what is k? can we either have a comment saying what they are, or else give them more descriptive names please?